backend/src/services/payments/payment.service.ts

```
@@ -0,0 +1,838 @@
+    environment: StripeEnvironment,
+    idempotencyKey: string | null | undefined,
+    task: () => Promise<T>
+  ): Promise<T> {
+    if (!idempotencyKey) {
+      return task();
+    }
+
+    return this.withSessionAdvisoryLock(
+      `payments_checkout_${environment}_${idempotencyKey}`,
+      task
+    );
```

ESLint

[error] Replace `⏎······`payments_checkout_${environment}_${idempotencyKey}`,⏎······task⏎····` with ``payments_checkout_${environment}_${idempotencyKey}`,·task`

(prettier/prettier)